avocado/plugins/run.py

```diff
@@ -166,9 +166,10 @@ def run(self, args):
             sys.exit(exit_codes.AVOCADO_FAIL)
         job_instance = job.Job(args)
         job_run = job_instance.run()
-        result_dispatcher = ResultDispatcher()
-        if result_dispatcher.extensions:
-            result_dispatcher.map_method('render',
-                                         job_instance.result,
-                                         job_instance)
+        if job_instance.result.tests_total > 0:
+            result_dispatcher = ResultDispatcher()
+            if result_dispatcher.extensions:
+                result_dispatcher.map_method('render',
+                                             job_instance.result,
+                                             job_instance)
```

Comment:
As I mentioned during our meeting, I don't think this is the way to go. I believe the result plugins should be able to deal with this.

Btw I also think we should expand the results API of job-error concept, which should be triggered when the job is about to finish with JOB_ERROR, with optional failure message. This would help in this case (we'd end up with JOB_ERROR + message about no tests) and the plugins could use their means to address that.

The difference is that right now the results plugins can only signal test-errors/failures/..., but they don't have any means of job-notes and job-errors. Most of the results plugins support such features (like testsuite.errors property in xunit)

Last think, empty testsuite is controversial topic. Python unittest allows it and reports PASS. We don't support it and there are many different opinions. I think we should not raise any failure in that case and we should generate the full results with no tests executed. (therefor the `job-errors` would not be used and the results_dispatcher would always run). Still implementing such method could be useful in other cases, like on discovery failure the `results.xml` could contain only the job-error and no tests (or if we already know the tests we could show all tests as skipped)